src/client/api/event.c

```diff
@@ -1218,6 +1218,7 @@ daos_event_priv_get(daos_event_t **ev)
 	if (evx->evx_status != DAOS_EVS_READY) {
 		D_CRIT("private event is inuse, status=%d\n",
 		       evx->evx_status);
+		D_ASSERT(0);
```

Comment:
Should probably just make this 
D_ASSERTF(evx->evx_status == DAOS_EVS_READY, ...);

On the other hand, since it's a client library, it may be better to return some error.   The user may want to do something before crashing.

Reply:
hmm i thought i removed this. i added it by trying to get a stack trace when trying to reproduce..